src/contexts/wallet.tsx

```diff
@@ -194,25 +215,116 @@ export const WalletProvider = ({ children = null as any }) => {
     }
   }
 
+  async function restore(
+    simulation: SorobanRpc.Api.SimulateTransactionRestoreResponse,
+    transaction: Transaction
+  ): Promise<ContractResponse<void>> {
+    let account = await rpc.getAccount(walletAddress);
+    // if (SorobanRpc.Api.isSimulationRestore(simulation)) {
+    setTxStatus(TxStatus.BUILDING);
+    let fee = parseInt(simulation.restorePreamble.minResourceFee) + parseInt(transaction.fee);
+    let restore_tx = new TransactionBuilder(account, { fee: fee.toString() })
+      .setNetworkPassphrase(network.passphrase)
+      .setSorobanData(simulation.restorePreamble.transactionData.build())
+      .addOperation(Operation.restoreFootprint({}))
+      .build();
+    let signed_restore_tx = new Transaction(await sign(restore_tx.toXDR()), network.passphrase);
+    setTxHash(signed_restore_tx.hash().toString('hex'));
+    let restore_tx_resp = await sendTransaction(signed_restore_tx, (result: string) => undefined);
+    if (restore_tx_resp) {
+      return restore_tx_resp;
+    } else {
+      throw new Error('Restore transaction failed');
+    }
+  }
+
+  async function sendTransaction<T>(
```

Comment:
naming on this one is a tad confusing , alex suggested something like :

```
submitTransaction
submitInvokeHostFunctionOperation
```

Reply:
yeah, it's a bit confusing between
`sendTransaction` and `submitTransaction`. The above might add a bit more clarity, but open to other ideas too.

Reply:
Yeah thought this was going to be confusing but wasn't sure what to put. I think the we do `InvokeOperation`  instead of submitTransaction and then keep sendTransaction

Reply:
`invokeSorobanOperation`?